### PR TITLE
Simplifications and re-factoring.

### PR DESCRIPTION
**Pull request by eric-ch**
- Autotools configuration can be simplified a lot using pkg-config and recent macros, most of the complexity in it was not used and could make it hard to follow.
- Generating exported headers with cproto tends to make things complicated without gaining much. It surely makes it harder to document things.
- Although this is a pretty simple library, some documentation can make things a bit easier.

Other minor changes include removal of auto-generated headers such, indentation glitches, .gitignore update, etc.

### CODE REVIEW

**Comment by jean-edouard** [comment]
This failed to build: http://158.69.127.133:8020/builders/openxt-custom/builds/29

**Comment by eric-ch** [comment]
My bad, it should be fixed now.

**Comment by jean-edouard** [comment]
Built here: http://158.69.127.133:8020/builders/openxt-custom/builds/33
It builds fine and I see no difference in the actual code.
Merging soon.
